dcc-portal-ui/app/scripts/pathwayviewer/js/services.js

```diff
@@ -716,95 +717,150 @@ angular.module('icgc.pathwayviewer.directives.services', [])
+              } else {
+                nodeValues[node.id] =  renderedValue;
+                highlighted.push(node.id);
+              }
+            });
+          });
+          
+          return nodeValues;
+        }
 
-        console.log('checking overlaps/highlights');
-
-        // Add SVG elements to nodes with highlight values
-        for (var nodeId in nodeValues) {
-          if (nodeValues.hasOwnProperty(nodeId)) {
+        function _drawAnnotations(annotations) {
+          var nodeValues = annotations.nodeValues;
+          var location = annotations.location;
+          var color = annotations.color;
+          
+          // Add SVG elements to nodes with highlight values
+          for (var nodeId in nodeValues) {
+            if (!(nodeValues.hasOwnProperty(nodeId))) {
+              continue;
+            }
 
-            var node = model.getNodeById(nodeId);
-            var renderedValue = nodeValues[nodeId];
-            var svgNode = svg.selectAll('.entity'+node.id);
-            svgNode.style('stroke',config.highlightColor);
+            var svgNode = svg.selectAll('.entity'+model.getNodeById(nodeId).id);
             svgNode.style('stroke-width','3px');
 
-            svg.append('rect')
-              .attr({
-                class:'value-banner value-banner'+nodeId,
-                x: (+node.position.x)+(+node.size.width) - 10,
-                y: (+node.position.y)- 7,
-                width:(renderedValue.toString().length*5)+10,
-                height:15,
-                rx: 7,
-                ry: 7,
-              }).style({
-              fill:config.highlightColor
+            var renderedValue = nodeValues[nodeId];
+            
+            // Draw rectangular container for annotation
+            svg.append('rect').attr({
+              class:'value-banner value-banner'+nodeId,
+              x: _coordinates(nodeId, location),
+              y: _coordinates(nodeId, 'top'),
+              width:(renderedValue.toString().length*5)+10,
+              height:15,
+              rx: 7,
+              ry: 7,
+            }).style({
+              fill:color
             });
 
+            // Draw annotation value inside container
             svg.append('text').attr({
               'class':'banner-text banner-text'+nodeId,
-              'x':(+node.position.x)+(+node.size.width) - 5,
-              'y':(+node.position.y)+4,
+              'x': _coordinates(nodeId, location, true),
+              'y': _coordinates(nodeId, 'top', true),
               'pointer-events':'none',
               'font-size':'9px',
               'font-weight':'bold',
               'fill':'white'
             }).text(renderedValue);
           }
+          
+          function _coordinates(nodeId, location, forText) {
+            var node = model.getNodeById(nodeId);
+            var renderedValue = nodeValues[nodeId];
+            
+            var coordinateFunctionMap = {
+              'left' : function(forText) {
```

Comment:
non-blocking minor nitpick, please don't quote object keys unless necessary (I'll make a linting rule for this once we switch to eslint)